src/commands/dev/dev.js

```diff
@@ -100,27 +108,47 @@ const runCommand = (command, env = {}) => {
 
   // we can't try->await->catch since we don't want to block on the framework server which
   // is a long running process
-  // eslint-disable-next-line promise/catch-or-return,promise/prefer-await-to-then
-  commandProcess.then(async () => {
-    const result = await commandProcess
-    const [commandWithoutArgs] = command.split(' ')
-    // eslint-disable-next-line promise/always-return
-    if (result.failed && isNonExistingCommandError({ command: commandWithoutArgs, error: result })) {
-      log(
-        NETLIFYDEVERR,
-        `Failed running command: ${command}. Please verify ${chalk.magenta(`'${commandWithoutArgs}'`)} exists`,
-      )
-    } else {
-      const errorMessage = result.failed
-        ? `${NETLIFYDEVERR} ${result.shortMessage}`
-        : `${NETLIFYDEVWARN} "${command}" exited with code ${result.exitCode}`
-
-      log(`${errorMessage}. Shutting down Netlify Dev server`)
-    }
-    process.exit(1)
-  })
+  // eslint-disable-next-line promise/catch-or-return
+  commandProcess
+    // eslint-disable-next-line promise/prefer-await-to-then
+    .then(async () => {
+      const result = await commandProcess
+      const [commandWithoutArgs] = command.split(' ')
+      // eslint-disable-next-line promise/always-return
+      if (result.failed && isNonExistingCommandError({ command: commandWithoutArgs, error: result })) {
+        log(
+          NETLIFYDEVERR,
+          `Failed running command: ${command}. Please verify ${chalk.magenta(`'${commandWithoutArgs}'`)} exists`,
+        )
+      } else {
+        const errorMessage = result.failed
+          ? `${NETLIFYDEVERR} ${result.shortMessage}`
+          : `${NETLIFYDEVWARN} "${command}" exited with code ${result.exitCode}`
+
+        log(`${errorMessage}. Shutting down Netlify Dev server`)
+      }
+    })
+    // eslint-disable-next-line promise/prefer-await-to-then
+    .then(() => {
+      // eslint-disable-next-line promise/always-return
+      if (!cleanupStarted) {
+        cleanupStarted = true
+        const cleanupFinished = Promise.all(cleanupWork.map((cleanup) => cleanup()))
+        return cleanupFinished
+      }
+    })
+    // eslint-disable-next-line promise/prefer-await-to-then,promise/always-return
+    .then(() => {
+      process.exit(1)
```

Comment:
I'm a little bit confused by the presence of this line - it exits too rapidly, and prevents the signal handlers below from actually completing currently. I've left it in looking for some guidance, since I'm not sure why it's here.